static/app/components/externalIssues/externalIssueForm.tsx

```diff
@@ -24,145 +50,300 @@ const SUBMIT_LABEL_BY_ACTION = {
   create: t('Create Issue'),
 };
 
-type Props = {
+interface ExternalIssueFormProps extends ModalRenderProps {
   group: Group;
   integration: Integration;
   onChange: (onSuccess?: () => void, onError?: () => void) => void;
-  organization: Organization;
-} & AbstractExternalIssueForm['props'];
+}
 
-type State = AbstractExternalIssueForm['state'];
+function makeIntegrationIssueConfigQueryKey({
+  orgSlug,
+  groupId,
+  integrationId,
+  action,
+}: {
+  groupId: string;
+  integrationId: string;
+  orgSlug: string;
+  action?: ExternalIssueAction;
+}): ApiQueryKey {
+  return [
+    `/organizations/${orgSlug}/issues/${groupId}/integrations/${integrationId}/`,
+    {query: {action}},
+  ];
+}
 
-export default class ExternalIssueForm extends AbstractExternalIssueForm<Props, State> {
-  loadSpan: Span | undefined;
-  submitSpan: Span | undefined;
-  trackedLoadStatus = false;
+export default function ExternalIssueForm({
+  group,
+  integration,
+  onChange,
+  closeModal,
+  Header,
+  Body,
+}: ExternalIssueFormProps) {
+  const api = useApi({persistInFlight: true});
+  const modelRef = useRef(new FormModel());
```

Comment:
I actually usually write these as setState, so we don't create a new FormModel every time

```suggestion
  const [formModel] = useState(() => new FormModel());
```

Reply:
that's clever! I'll make the change